src/main/java/com/sublinks/sublinksapi/person/services/UserDataService.java

```diff
@@ -37,45 +37,20 @@ public void checkAndAddIpRelation(Person person, String ipAddress, String token,
       @Nullable String userAgent) {
 
     boolean saveUserIps = userDataConfig.isSaveUserData();
-    Optional<PersonMetaData> foundData = getActiveUserDataByPersonAndIpAddress(person, token, ipAddress,
-        userAgent);
+    Optional<PersonMetaData> foundData = getActiveUserDataByPersonAndToken(person, token);
 
     if (foundData.isPresent()) {
 
       PersonMetaData personMetaData = foundData.get();
-
-      if (saveUserIps && userAgent != null && !personMetaData.getUserAgent().equals(userAgent)) {
-        final PersonMetaData newPersonMetaData = PersonMetaData.builder()
-            .person(person)
-            .ipAddress(ipAddress)
-            .userAgent(userAgent)
-            .token(token)
-            .active(true)
-            .build();
-
-        userDataRepository.save(newPersonMetaData);
-        userDataCreatedEventPublisher.publish(newPersonMetaData);
-        return;
-      }
-
-      if (saveUserIps && !personMetaData.getIpAddress().equals(ipAddress)) {
-        final PersonMetaData newPersonMetaData = PersonMetaData.builder()
-            .person(person)
-            .ipAddress(ipAddress)
-            .userAgent(userAgent)
-            .token(token)
-            .active(true)
-            .build();
-
-        userDataRepository.save(newPersonMetaData);
-        userDataCreatedEventPublisher.publish(newPersonMetaData);
-        return;
+      if (saveUserIps) {
+        personMetaData.setIpAddress(ipAddress);
+        if (userAgent != null) {
```

Comment:
Wouldn't we want to save the userAgent regardless of the saveUserIps setting?

Reply:
SaveUserIps is probably wrongly named rather dont save an userdata

Reply:
Will rename it later